CalibTracker/SiStripESProducers/plugins/real/SiStripBackPlaneCorrectionDepESProducer.cc

```diff
@@ -61,11 +61,11 @@ std::shared_ptr<SiStripBackPlaneCorrection> SiStripBackPlaneCorrectionDepESProdu
   if ( backPlaneCorrectionRecordName == "SiStripBackPlaneCorrectionRcd"){
     edm::ESHandle<SiStripBackPlaneCorrection> siStripBackPlaneCorrection;
     iRecord.getRecord<SiStripBackPlaneCorrectionRcd>().get(backPlaneCorrectionLabel, siStripBackPlaneCorrection);
-    siStripBPC_.reset(new SiStripBackPlaneCorrection(*(siStripBackPlaneCorrection.product())));
+    siStripBPC.reset(new SiStripBackPlaneCorrection(*(siStripBackPlaneCorrection.product())));
```

Comment:
@gartung I guess I am becoming annoying, but I believe it would be better to write this:
```
siStripBPC = std::make_shared<SiStripBackPlaneCorrection>(*(siStripBackPlaneCorrection.product()));
```
If I am not mistaken this is preferred because of exception safety. Alternatively one could also write this:
```
siStripBPC.reset(std::make_shared<SiStripBackPlaneCorrection>(*(siStripBackPlaneCorrection.product())));
```
@Dr15Jones since this PR is basically about thread-safety, we should also consider to make it more exception-safe, right?

Reply:
I'm pretty certain the `siStripBPC.reset` passed a `std::shared_ptr` would not compile, since `reset` expects a pointer. I do agree that the `operator=` version would be better.

Reply:
@Dr15Jones regarding the `reset`, I must admit that I did not check the documentation, but you are right, of course